### PR TITLE
fix: do not hide conversation when its content is cleared

### PR DESCRIPTION
**Pull request by PatrykBuniX**
## Description

According to spec, conversation should not be gone from the conversations list after its content gets cleared. "Clear content" button should only clear the contents and do nothing more.

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/conversation/ConversationRepository.ts [hunk 1]
```diff
@@ -2134,24 +2134,15 @@ export class ConversationRepository {
...
-
+  public async clearConversationContent(conversation: Conversation) {
     await this.messageRepository.updateClearedTimestamp(conversation);
-    await this._clearConversation(conversation);
-
-    if (isActiveConversation) {
-      amplify.publish(WebAppEvents.CONVERSATION.SHOW, nextConversationEntity, {});
-    }
+    await this.deleteMessages(conversation);
+    return this.addCreationMessage(conversation, !!this.userState.self()?.isTemporaryGuest());
```
Comment (PatrykBuniX) [inline src/script/conversation/ConversationRepository.ts]: We still want the creation message to be in the conversation.

File: src/script/conversation/ConversationRepository.ts [hunk 2]
```diff
@@ -3710,6 +3700,7 @@ export class ConversationRepository {
     conversationEntity.hasCreationMessage = false;
 
     const iso_date = timestamp ? new Date(timestamp).toISOString() : undefined;
+    conversationEntity.removeMessages();
```
Comment (PatrykBuniX) [inline src/script/conversation/ConversationRepository.ts:3700]: Removing messages also from the local state (not only from the indexedDB)

File: src/script/conversation/ConversationState.ts [hunk 3]
```diff
@@ -84,7 +84,6 @@ export class ConversationState {
     this.visibleConversations = ko.pureComputed(() => {
       return this.sortedConversations().filter(
         conversation =>
-          !conversation.is_cleared() &&
```
Comment (PatrykBuniX) [inline src/script/conversation/ConversationState.ts:87]: From this point we don't want to hide cleared conversations.

**Review by V-Gira — APPROVED** [review]

File: src/script/conversation/ConversationRepository.ts [hunk 4]
```diff
@@ -2463,7 +2453,7 @@ export class ConversationRepository {
   /**
    * Clears conversation content from view and the database.
    *
-   * @param conversationEntity Conversation entity to delete
+   * @param conversationEntity Conversation entity to clear
    * @param timestamp Optional timestamps for which messages to remove
    */
   private async _clearConversation(conversationEntity: Conversation, timestamp?: number) {
```
Comment (atomrc) [inline src/script/conversation/ConversationRepository.ts]: Shouldn't this one also be adjusted as well?

Comment (PatrykBuniX) [reply src/script/conversation/ConversationRepository.ts]: Good catch, I've ended up removing this method and creating `onConversationContentCleared` that is fired when we receive member update event in self-conversation with information that conversation was cleared, so all other self clients also do the "clear". https://github.com/wireapp/wire-webapp/pull/16247/commits/16cee22b6e3014c11bd048e59313065e4902b291

**Review by atomrc — APPROVED** [review]

**Pull request merged**